cmd/net_test.go

```diff
@@ -291,10 +291,9 @@ func TestSameLocalAddrs(t *testing.T) {
 		expectedErr error
 	}{
 		{"", "", false, errors.New("unable to process empty address")},
-		{":9000", ":9000", true, nil},
-		{"localhost:9000", ":9000", true, nil},
-		{"localhost:9000", "http://localhost:9000", true, nil},
-		{"8.8.8.8:9000", "http://localhost:9000", false, nil},
+		{"http://localhost:9000", ":9000", true, nil},
+		{"http://localhost:9000", "http://localhost:9000", true, nil},
+		{"http://8.8.8.8:9000", "http://localhost:9000", false, nil},
```

Comment:
As `extractHostPort()` is used in `sameLocalAddrs()` and `extractHostPort()` supports three types ":9000", "localhost:9000", "http://localhost:9000/".

I see `sameLocalAddrs()` is used in gateway-main.go and values from command line arguments are passed.  There is a possibility of three types of value passed to the function.

I think you need to keep previous cases in addition.